dd4t-core/src/main/java/org/dd4t/core/factories/impl/PageFactoryImpl.java

```diff
@@ -41,6 +41,8 @@
 	private static final Logger LOG = LoggerFactory.getLogger(PageFactoryImpl.class);
 	private static final PageFactoryImpl INSTANCE = new PageFactoryImpl();
 
+	private static final Object CACHE_LOCK  = new Object();

```

Comment:
Local variables cannot be used for synchronized locks since they're unique for every single thread. To really have a synchronization we need something common; and of of common ways is to have static LOCK object in a class so that it will be shared between threads.